serial.WriteBigInt: nil will throw

```go
func WriteBigInt(w io.Writer, i *big.Int, err *error) {
       // i may be nil
	Write(w, i.Bytes(), err)
}
```

```plain
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xc0000005 code=0x0 addr=0x10 pc=0x890d5d]

goroutine 52 [running]:
testing.tRunner.func1.1(0xe34f80, 0xd4c6e0)
	C:/Go/src/testing/testing.go:941 +0x3d7
testing.tRunner.func1(0xc00097c000)
	C:/Go/src/testing/testing.go:944 +0x400
panic(0xe34f80, 0xd4c6e0)
	C:/Go/src/runtime/panic.go:967 +0x16b
github.com/golang/mock/gomock.(*Controller).Finish(0xc0009705a0)
	C:/work/go/pkg/mod/github.com/golang/mock@v1.4.0/gomock/controller.go:246 +0x281
panic(0xe34f80, 0xd4c6e0)
	C:/Go/src/runtime/panic.go:967 +0x16b
math/big.(*Int).Bytes(...)
	C:/Go/src/math/big/int.go:451
github.com/HyperService-Consortium/go-uip/lib/serial.WriteBigInt(0x10ad120, 0xc000970630, 0x0, 0xc00091bca0)
	C:/work/go/pkg/mod/github.com/!hyper!service-!consortium/go-uip@v0.0.0-20200408075657-d2425491ab24/lib/serial/io.go:78 +0x2d
github.com/HyperService-Consortium/go-uip/internal/lexer_types.(*Uint256).Marshal(0x0, 0x10ad120, 0xc000970630, 0xc00091bca0)
	C:/work/go/pkg/mod/github.com/!hyper!service-!consortium/go-uip@v0.0.0-20200408075657-d2425491ab24/internal/lexer_types/vtok-const.go:82 +0x50
github.com/HyperService-Consortium/go-uip/internal/lexer_types.EncodeVTok(0x10ad120, 0xc000970630, 0x10ca0e0, 0x0, 0xc00091bca0)
	C:/work/go/pkg/mod/github.com/!hyper!service-!consortium/go-uip@v0.0.0-20200408075657-d2425491ab24/internal/lexer_types/serial.go:24 +0xdf
github.com/HyperService-Consortium/go-uip/op-intent.EncodeVTok(...)
	C:/work/go/pkg/mod/github.com/!hyper!service-!consortium/go-uip@v0.0.0-20200408075657-d2425491ab24/op-intent/serializer.go:15
github.com/HyperService-Consortium/go-ves/dependency/uip.(*ContractMetaEncoder).Marshal(0x0, 0xc00007de28, 0xc0009579d0, 0xc00011ab00, 0x7b, 0x7b, 0x4f5f04)
	C:/work/go/src/github.com/HyperService-Consortium/go-ves/dependency/uip/intents.go:20 +0x1f7
github.com/HyperService-Consortium/go-ves/ves/service/session.Test_prepareTranslateEnvironment_doContractInvoke(0xc00097c000)
	C:/work/go/src/github.com/HyperService-Consortium/go-ves/ves/service/session/prepare_transaction_environment_test.go:262 +0x350
testing.tRunner(0xc00097c000, 0xf91850)
	C:/Go/src/testing/testing.go:992 +0xe3
created by testing.(*T).Run
	C:/Go/src/testing/testing.go:1043 +0x35e
```

--- a/lib/serial/io.go
+++ b/lib/serial/io.go
@@ -75,6 +75,10 @@ func EncodeBigInt(b *big.Int) ([]byte, error) {
 }
 
 func WriteBigInt(w io.Writer, i *big.Int, err *error) {
+	if i == nil {
+		Write(w, []byte(nil), err)
+		return
+	}
 	Write(w, i.Bytes(), err)
 }
 

--- a/lib/serial/io_test.go
+++ b/lib/serial/io_test.go
@@ -1,0 +1,71 @@
+package serial
+
+import (
+	"bytes"
+	"github.com/stretchr/testify/assert"
+	"io"
+	"math/big"
+	"strings"
+	"testing"
+)
+
+func TestWriteBigInt(t *testing.T) {
+	type args struct {
+		i   *big.Int
+		err *error
+	}
+	tests := []struct {
+		name  string
+		args  args
+		wantW string
+	}{
+		{"normal", args{big.NewInt(1), new(error)},
+			"\x00\x00\x00\x00\x00\x00\x00\x01" + string(big.NewInt(1).Bytes())},
+		{"nil", args{nil, new(error)},
+			"\x00\x00\x00\x00\x00\x00\x00\x00" + string(big.NewInt(1).SetBytes(nil).Bytes())},
+		{"nil", args{big.NewInt(0).SetBytes(nil), new(error)},
+			"\x00\x00\x00\x00\x00\x00\x00\x00" + string(big.NewInt(1).SetBytes(nil).Bytes())},
+		{"nil", args{big.NewInt(1).SetBytes(nil), new(error)},
+			"\x00\x00\x00\x00\x00\x00\x00\x00" + string(big.NewInt(1).SetBytes(nil).Bytes())},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			w := &bytes.Buffer{}
+			WriteBigInt(w, tt.args.i, tt.args.err)
+			if *tt.args.err != nil {
+				t.Errorf("write error: %v", *tt.args.err)
+			}
+			if gotW := w.String(); gotW != tt.wantW {
+				t.Errorf("WriteBigInt() = %v, want %v", []byte(gotW), []byte(tt.wantW))
+			}
+		})
+	}
+}
+
+func TestReadBigInt(t *testing.T) {
+	type args struct {
+		r   io.Reader
+		i   *big.Int
+		err *error
+	}
+	tests := []struct {
+		name  string
+		args  args
+		wantI *big.Int
+	}{
+		{"normal",
+			args{strings.NewReader("\x00\x00\x00\x00\x00\x00\x00\x01" + string(big.NewInt(1).Bytes())),
+				big.NewInt(0), new(error)}, big.NewInt(1)},
+		{"nil", args{strings.NewReader("\x00\x00\x00\x00\x00\x00\x00\x00" + string(big.NewInt(1).SetBytes(nil).Bytes())),
+			big.NewInt(0), new(error)}, big.NewInt(0).SetBytes([]byte{})},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			ReadBigInt(tt.args.r, tt.args.i, tt.args.err)
+			if *tt.args.err != nil {
+				t.Errorf("read error: %v", *tt.args.err)
+			}
+			assert.EqualValues(t, tt.wantI, tt.args.i)
+		})
+	}
+}

